File: cses/general_weird_algorithm.py
n = int(input())
res = []
while n != 1:
    res.append(int(n))
    if n % 2 == 0:
        n /= 2
    else:
        n = n * 3 + 1
if len(res) == 0 or res[-1] != 1:
    res.append(1)
res = [str(x) for x in res]
print(' '.join(res))

File: cses/general_weird_algorithm.cpp
#include "vector"
#include "iostream"
#include "sstream"
#include "string"

int	main()
{
	std::vector<long long>	res;
	long long		n;
	int			i;

	std::cin >> n;
	while (n != 1)
	{
		res.push_back(n);
		if (n % 2)
			n = 3 * n + 1;
		else
			n /= 2;
	}
	if ((int) res.size() == 0 || res.back() != 1)
		res.push_back(1);
	i = 0;
	std::cout << res[i];
	while (++i < (int) res.size())
	{
		std::cout << ' ' << res[i];
	}
	return (0);
}
